homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -261,12 +279,85 @@ async def async_step_set_device_options(self, user_input=None):
                 }
             )
 
+        if isinstance(self._selected_device_object, rfxtrxmod.SensorEvent):
+            devices = {
+                entry.id: entry.name_by_user if entry.name_by_user else entry.name
+                for entry in self._device_entries
+                if self._can_replace_device(entry.id)
+            }
+
+            if devices:
+                data_schema.update(
+                    {
+                        vol.Optional(CONF_REPLACE_DEVICE): vol.In(devices),
+                    }
+                )
+
         return self.async_show_form(
             step_id="set_device_options",
             data_schema=vol.Schema(data_schema),
             errors=errors,
         )
 
+    async def _async_replace_device(self, replace_device):
+        """Migrate properties of a device into another."""
+        device_registry = self._device_registry
+        old_device_id = self._selected_device_entry_id
+        old_entry = device_registry.async_get(old_device_id)
+        device_registry.async_update_device(
+            replace_device,
+            area_id=old_entry.area_id,
+            name_by_user=old_entry.name_by_user,
+        )
+
+        entity_registry = await async_get_entity_registry(self.hass)
+        entity_migration_map = {}
+        for entry in entity_registry.entities.values():
+            if entry.device_id == replace_device and entry.domain == SENSOR_DOMAIN:
```

Comment:
> Why is it only sensor entities that are relevant? What if there will be new domains for this type of device in the future?

I currently made it such that the migration function is only presented for event codes that in the underlying library result in a `SensorPacket`. These packets in the current implementation only use sensor platform. The reason to exclude other domains is that the logic is a bit different because sensor entities have their unique id prepended with a data type.

There is nothing against adding other domains, but we could as well then also provide migration function for all type of data packets. I only know about sensors with behavior prescribed in the PR description, but there may be e.g. doorbells which do the same.

> Isn't it enough to check the device id and migrate all entity entries for the device?

Wouldn't we want to also migrate the user defined device name from the old to new device?

Reply:
We want to migrate device name and area too. I just meant that we didn't need a check for domain or unique_id for the entitie. Just migrate all entity entriesffor tthe decide by defaul.

Reply:
When a device has multiple entities, what would be the best way to match entities from old/new device? I was using the domain/unique id for that.

Reply:
Yes. I was thinking, if we, for each old entity entry for a device replace the device id in the unique id with the new device id, we should get the new unique id. Then we can look up the new entry entity id with this helper:
https://github.com/home-assistant/core/blob/c64eec3238c577d6e9f1c184c4d2ae0b6e711fef/homeassistant/helpers/entity_registry.py#L159-L162

The domain and platform will be the same for the old and new entry.

Will that work?

Reply:
Yes, we could get entities with `async_entries_for_device` for the old device and replace the device id in unique id. That would work for all domains.

Reply:
Changed it to above methods. Also removed the check for `SensorPacket` so migration should work for all device types. Will add a test for that as well